imageregistry/v1/types.go

```diff
@@ -82,6 +82,10 @@ type ImageRegistrySpec struct {
 	// tolerations defines the tolerations for the registry pod.
 	// +optional
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty" protobuf:"bytes,14,rep,name=tolerations"`
+	// rolloutStrategy defines rollout strategy for the image registry
+	// deployment.
+	// +optional
+	RolloutStrategy string `json:"rolloutStrategy" protobuf:"bytes,15,opt,name=rolloutStrategy"`
```

Comment:
@sttts can we make this a `appsv1.DeploymentStrategyType` [1]? Would we automatically get the validation pattern we want?

If not, then the validation pattern needs to be added as a comment tag - see [2] as an example.

[1] https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L335
[2] https://github.com/openshift/api/blob/master/imageregistry/v1/types.go#L239

Reply:
done!

Reply:
@adambkaplan upstream types are coded (and validated) in Go. There is no connection to these validation here and those in Go land.